packages/react-slider/src/components/Slider/useSliderState.tsx

```diff
@@ -167,16 +157,10 @@ export const useSliderState = (state: SliderState) => {
     return origin !== undefined ? getPercent(origin, min, max) : 0;
   }, [max, min, origin]);
 
-  const markValues = React.useMemo((): number[] => getMarkValue(marks, min, max, step), [marks, max, min, step]);
-
-  const markPercent = React.useMemo((): string[] => getMarkPercent(markValues), [markValues]);
-
-  const thumbWrapperStyles = {
-    transform: vertical
-      ? `translateY(${valuePercent}%)`
-      : `translateX(${dir === 'rtl' ? -valuePercent : valuePercent}%)`,
-    transition: stepAnimation ? `transform ease-in-out ${animationTime}` : 'none',
-    ...state.thumbWrapper.style,
+  const thumbStyles = {
```

Comment:
not blocking: I think we should move this to the `useSliderStyles`  hook ?

Reply:
might be worth moving the state based styles into a css variable, and letting styles pick those up